c7n/resources/vpc.py

```diff
@@ -878,12 +879,41 @@ class UsedSecurityGroup(SGUsage):
     """
     schema = type_schema('used')
 
+    instance_owner_id_key = 'c7n:InstanceOwnerIds'
+    interface_type_key = 'c7n:InterfaceTypes'
+
+    def _set_eni_attributes(self):
+        enis = []
+        for nic in self.nics:
+            if nic['Status'] == 'in-use':
+                instance_owner_id = nic['Attachment']['InstanceOwnerId']
+            else:
+                instance_owner_id = ''
+            interface_type = nic.get('InterfaceType')
+            for g in nic['Groups']:
+                enis.append({'GroupId': g['GroupId'],
+                             'InstanceOwnerId': instance_owner_id,
+                             'InterfaceType': interface_type})
+        return enis
+
     def process(self, resources, event=None):
         used = self.scan_groups()
         unused = [
             r for r in resources
             if r['GroupId'] not in used and 'VpcId' in r]
         unused = {g['GroupId'] for g in self.filter_peered_refs(unused)}
+        enis = self._set_eni_attributes()
+        for r in resources:
+            r[self.instance_owner_id_key] = []
+            r[self.interface_type_key] = []
+            for i in range(1, len(enis)):
+                if r['GroupId'] == enis[i]['GroupId']:
+                    r[self.instance_owner_id_key].append(enis[i]['InstanceOwnerId'])
+                    r[self.interface_type_key].append(enis[i]['InterfaceType'])
+            r[self.instance_owner_id_key] = list(filter(None, set(
```

Comment:
This feels like an awkward combination of methods. If the idea is to end up with a list of unique owner IDs as the resource annotation, one alternative might be:

```python
# start with a set so you don't have to deduplicate later
owner_ids = set()
# inside the loop, add owners
owner_ids.add(eni['InstanceOwnerId'])
# convert back to a list at the end
r[self.instance_owner_id_key] = list(owner_ids)
```

If that does the trick, a similar approach could work for interface types.

Reply:
Fixed. I added `set` and `filter` afterward and never come back to check the code. Thanks for catching it.